test/src/kernels/DeprecatedCoupledVarKernel.C

```diff
@@ -0,0 +1,35 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "DeprecatedCoupledVarKernel.h"
+
+registerMooseObject("MooseTestApp", DeprecatedCoupledVarKernel);
+
+InputParameters
+DeprecatedCoupledVarKernel::validParams()
+{
+  InputParameters params = ADKernel::validParams();
+  params.addCoupledVar("source", "A variable representing a source term");
+  params.addDeprecatedCoupledVar("stupid_name",
+                                 "source",
+                                 "A variable representing a source term",
+                                 "'stupid_name' is deprecated. Please use 'source' instead");
```

Comment:
This is fine, but I'd think that the last two arguments will just be boilerplate. They could be automatically generated. But if you think the additional flexibility is necessary I'm ok with that, too.

Reply:
This reviewer drives a hard bargain! I like your idea though. I'll make the change